x-pack/plugins/siem/public/components/super_date_picker/index.tsx

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import dateMath from '@elastic/datemath';
+import {
+  EuiSuperDatePicker,
+  EuiSuperDatePickerProps,
+  OnRefreshChangeProps,
+  OnRefreshProps,
+  OnTimeChangeProps,
+} from '@elastic/eui';
+import { get, getOr } from 'lodash/fp';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { ActionCreator } from 'typescript-fsa';
+
+import { inputsActions, inputsModel, State } from '../../store';
+
+type MyEuiSuperDatePickerProps = Pick<
+  EuiSuperDatePickerProps,
+  | 'end'
+  | 'isPaused'
+  | 'onTimeChange'
+  | 'onRefreshChange'
+  | 'onRefresh'
+  | 'recentlyUsedRanges'
+  | 'refreshInterval'
+  | 'showUpdateButton'
+  | 'start'
+> & {
+  isLoading?: boolean;
+};
+const MyEuiSuperDatePicker: React.SFC<MyEuiSuperDatePickerProps> = EuiSuperDatePicker;
+
+interface SuperDatePickerStateRedux {
+  duration: number;
+  kind: string;
+  option: string;
+  start: number;
+  end: number;
+  isLoading: boolean;
+  refetch: inputsModel.Refetch[];
+}
+
+interface SuperDatePickerDispatchProps {
+  setAbsoluteSuperDatePicker: ActionCreator<{ id: string; from: number; to: number }>;
+  setRelativeSuperDatePicker: ActionCreator<{
+    id: string;
+    option: string;
+    from: number;
+    to: number;
+  }>;
+  startAutoReload: ActionCreator<{ id: string }>;
+  stopAutoReload: ActionCreator<{ id: string }>;
+  setDuration: ActionCreator<{ id: string; duration: number }>;
+}
+interface OwnProps {
+  id: string;
+  disabled?: boolean;
+}
+
+interface TimeArgs {
+  start: string;
+  end: string;
+}
+
+export type SuperDatePickerProps = OwnProps &
+  SuperDatePickerDispatchProps &
+  SuperDatePickerStateRedux;
+
+export interface SuperDatePickerState {
+  isAutoRefreshOnly: boolean;
+  isPaused: boolean;
+  isQuickSelection: boolean;
+  recentlyUsedRanges: TimeArgs[];
+  refreshInterval: number;
+  showUpdateButton: boolean;
+}
+
+export const SuperDatePickerComponent = class extends Component<
+  SuperDatePickerProps,
+  SuperDatePickerState
+> {
+  constructor(props: SuperDatePickerProps) {
+    super(props);
+
+    this.state = {
+      isAutoRefreshOnly: false,
+      isPaused: true,
+      isQuickSelection: true,
+      recentlyUsedRanges: [],
+      refreshInterval: 300000,
+      showUpdateButton: true,
+    };
+  }
+
+  public render() {
+    const { end, start, kind, option, isLoading } = this.props;
+    let endDate = new Date(end).toISOString();
```

Comment:
For consistency throughout the codebase, where we prefer `const` over `let`, consider the following refactoring:

```
    const endDate = kind === 'relative' ? 'now' : new Date(end).toISOString();
    const startDate = kind === 'relative' ? option : new Date(start).toISOString();
```